src/fdb5/api/local/ListVisitor.h

```diff
@@ -55,6 +55,13 @@ struct ListVisitor : public QueryVisitor<ListElement> {
             indexRequest_.unsetValues(kv.first);
         }
 
+        if (level_ == 1) {
+            /// @todo fix this so that we don't need to create an empty key
+            const auto emptyKey = Key {currentCatalogue_->schema().registry()};
```

Comment:
Is this necessary with the schema().registry() - given that there are no keys contained inside? Does anything break if we just default construct the empty keys().

Further, can we not just emplace a single part key here, rather than all 3 parts?

Reply:
we need this hacky empty keys since there's a Key hash functor using valuesToString .. which kicks in when Key used in unordered containers.. see the note, I will attempt to fix in a later PR.